From 74cf53d89529902ecdef5b230421bc4d30fd089b Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 25 Jan 2023 15:36:14 -0800
Subject: [PATCH] include missing <cstdint>

gcc-13 exposed errors like below due to it removing indirect includes of
these headers in libstdc++ [1]

../git/external/amber/src/include/amber/shader_info.h:60:15: error: 'uint32_t' was not declared in this scope
   60 |   std::vector<uint32_t> shader_data;
      |               ^~~~~~~~

[1] https://www.gnu.org/software/gcc/gcc-13/porting_to.html

Upstream-Status: Submitted [https://github.com/google/amber/pull/1016]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 include/amber/recipe.h                  | 1 +
 include/amber/shader_info.h             | 1 +
 src/descriptor_set_and_binding_parser.h | 1 +
 3 files changed, 3 insertions(+)

diff --git a/include/amber/recipe.h b/include/amber/recipe.h
index 310a86a..1b83213 100644
--- a/include/amber/recipe.h
+++ b/include/amber/recipe.h
@@ -15,6 +15,7 @@
 #ifndef AMBER_RECIPE_H_
 #define AMBER_RECIPE_H_
 
+#include <cstdint>
 #include <string>
 #include <utility>
 #include <vector>
diff --git a/include/amber/shader_info.h b/include/amber/shader_info.h
index 6c9126a..b5ce751 100644
--- a/include/amber/shader_info.h
+++ b/include/amber/shader_info.h
@@ -15,6 +15,7 @@
 #ifndef AMBER_SHADER_INFO_H_
 #define AMBER_SHADER_INFO_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/descriptor_set_and_binding_parser.h b/src/descriptor_set_and_binding_parser.h
index 145aa42..4b0d774 100644
--- a/src/descriptor_set_and_binding_parser.h
+++ b/src/descriptor_set_and_binding_parser.h
@@ -15,6 +15,7 @@
 #ifndef SRC_DESCRIPTOR_SET_AND_BINDING_PARSER_H_
 #define SRC_DESCRIPTOR_SET_AND_BINDING_PARSER_H_
 
+#include <cstdint>
 #include <string>
 
 #include "amber/result.h"
-- 
2.39.1

